frontend/src/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -6,6 +6,7 @@ interface ComponentProps {
   options: DropdownItemProps[];
   selectedOptions?: string[];
   onUpdateOptions?: (options: string) => void;
+  validateCheckboxOptions?: (option: string, checkboxState: boolean | undefined) => void;
```

Comment:
I'm pretty sure you only care about the deselection of a checkbox, so lets rename this to `onDeselect` and remove `checkboxState` ... You'll just have to make sure you only call this if the checkbox is deselected.
Also, there's a much better way of writing `checkboxState: boolean | undefined`, without the `| undefined`

Reply:
I also want the name to not be about validation, cos I may want to use this nice `CheckboxGroup` component outside of the QB, and do something else on deselect.

Reply:
Renamed to `onDeselect` and is only called when checkbox is deselected. Also removed `checkboxState`